File: IF Quest/Em_python/Guerreiro.py
from Personagem import Personagem

class Guerreiro(Personagem):
    def __init__(self, nome: str, vida: int, nivel: int, forca: int):
        super().__init__(nome, vida, nivel, forca)
        self._defesa = 5

    def habilidade(self) -> str:
        return "corte_duplo"

    def atacar(self, alvo=None):
        super().atacar(alvo)

    @property
    def defesa(self):
        return self._defesa

    @defesa.setter
    def defesa(self, valor: int):
        if valor < 0:
            raise ValueError("A defesa não pode ser negativa.")
        else:
            self._defesa = valor

    def ficha(self):
        print("Ficha do Guerreiro:")
        super().ficha()
        print(f"Defesa: {self._defesa}")

File: IF Quest/Em_java/Guerreiro.java
public class Mago extends Personagem{
    private int mana;
    public Mago(String nome, int vida, int nivel, int forca){
        super(nome, vida, nivel, forca);
        this.setMana(50);
    }
    public int getMana(){
        return mana;
    }
    public void setMana(int mana){
        if(mana < 0){
            throw new IllegalArgumentException("Mana nao pode ser negativa: " + mana);
        }
        this.mana = mana;
    }
    @Override
    public String habilidade(){
        return "Expelliarmus";
    }
    @Override
    public void atacar(Personagem alvo) throws SemMana{
        if(this.mana < 10) throw new SemMana(this.mana);
        this.mana -= 10;
        super.atacar(alvo);
    }
    @Override
    public void ficha(){
        System.out.println("\nMago");
        super.ficha();
        System.out.println("Mana: " + this.mana);
    }
}
